services/user/Workshop/service/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+#[psibase::service]
+#[allow(non_snake_case)]
+pub mod service {
+    use async_graphql::*;
+    use psibase::*;
+    use serde::{Deserialize, Serialize};
+
+    /// Holds metadata for a registered app
+    #[table(name = "AppMetadataTable", index = 0)]
+    #[derive(Debug, Clone, Fracpack, ToSchema, Serialize, Deserialize, SimpleObject)]
+    pub struct AppMetadata {
+        /// The unique identifier for the app
+        #[primary_key]
+        account_id: AccountNumber,
+
+        /// The name of the app
+        name: String,
+
+        /// A short description of the app
+        short_description: String,
+
+        /// A detailed description of the app
+        long_description: String,
+
+        /// The icon of the app (stored as a base64 string)
+        icon: Option<String>,
+
+        /// The subpage for Terms of Service
+        tos_subpage: String,
+
+        /// The subpage for Privacy Policy
+        privacy_policy_subpage: String,
+
+        /// The subpage for the app's homepage
+        app_homepage_subpage: String,
+
+        /// The status of the app (DRAFT, PUBLISHED, or UNPUBLISHED)
+        status: String, // todo: change to enum
+    }
+
+    #[table(record = "WebContentRow", index = 1)]
+    pub struct WebContentTable;
+
+    #[action]
+    pub fn setAppMetadata(
+        name: String,
+        short_description: String,
+        long_description: String,
+        icon: Option<String>, // todo: add hex and type
+        tos_subpage: String,
+        privacy_policy_subpage: String,
+        app_homepage_subpage: String,
+        status: String,
+    ) {
+        let app_metadata_table = AppMetadataTable::new();
+        let account_id = get_sender();
+
+        let app_metadata = AppMetadata {
+            account_id,
+            name,
+            short_description,
+            long_description,
+            icon,
+            tos_subpage,
+            privacy_policy_subpage,
+            app_homepage_subpage,
+            status,
+        };
+
+        app_metadata_table.put(&app_metadata).unwrap();
+
+        Wrapper::emit().history().setAppMetadata(app_metadata);
+    }
+
+    #[action]
+    fn storeSys(path: String, contentType: String, content: Hex<Vec<u8>>) {
```

Comment:
`storeSys` (and therefore the `WebContentTable`) should be in the query service alongside `serveSys`.

Reply:
when I read query my first instinct was that it is a read only service (I thought the same when I saw the example named `r-<package>`). If we are adding the storesys it would go against a query or a read only service. So what is the definition of the query service on the package?

Reply:
A query service's primary job is to handle HTTP requests. In this mode it is read-only. The data is stored in the query service because it is only used for the HTTP interface. It's essentially static content associated with the service.

Reply:
I think I might prefer calling it an "http-server" or something. Query / RPC service don't seem as readily understandable.